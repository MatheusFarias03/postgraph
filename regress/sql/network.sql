/*
 * Copyright (C) 2023 PostGraphDB
 *  
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *  
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

LOAD 'postgraph';
SET search_path TO postgraph;

SELECT create_graph('network');

--
-- inet
--
SELECT * FROM cypher('network', $$ RETURN toinet('192.168.1.5') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN toinet('192.168.1/24') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN toinet('::ffff:fff0:1') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '192.168.1.5'::inet $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '192.168.1/24'::inet $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '::ffff:fff0:1'::inet $$) as (i gtype);


--
-- cidr
--
SELECT * FROM cypher('network', $$ RETURN tocidr('192.168.1.5') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN tocidr('192.168.1/24') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN tocidr('::ffff:fff0:1') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '192.168.1.5'::cidr $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '192.168.1/24'::cidr $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '::ffff:fff0:1'::cidr $$) as (i gtype);

--
-- macaddr
--
SELECT * FROM cypher('network', $$ RETURN tomacaddr('12:34:56:78:90:ab') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '12:34:56:78:90:ab'::macaddr $$) as (i gtype);

--
-- macaddr8
--
SELECT * FROM cypher('network', $$ RETURN tomacaddr8('12:34:56:78:90:ab:cd:ef') $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN '12:34:56:78:90:ab:cd:ef'::macaddr8 $$) as (i gtype);

--
-- abbrev
--
SELECT * FROM cypher('network', $$ RETURN abbrev('192.168.1.5'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('192.168.1/24'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('::ffff:fff0:1'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('10.1.0.0/32'::inet) $$) as (i gtype);

SELECT * FROM cypher('network', $$ RETURN abbrev('192.168.1.5'::cidr) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('192.168.1/24'::cidr) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('::ffff:fff0:1'::cidr) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('10.1.0.0/32'::cidr) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN abbrev('10.1.0.0/16'::cidr) $$) as (i gtype);

--
-- broadcast
--
SELECT * FROM cypher('network', $$ RETURN broadcast('192.168.1.5'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN broadcast('192.168.1/24'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN broadcast('::ffff:fff0:1'::inet) $$) as (i gtype);
SELECT * FROM cypher('network', $$ RETURN broadcast('10.1.0.0/32'::inet) $$) as (i gtype);


SELECT drop_graph('network', true);
