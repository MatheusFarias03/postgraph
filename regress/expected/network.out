/*
 * Copyright (C) 2023 PostGraphDB
 *  
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *  
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
LOAD 'postgraph';
SET search_path TO postgraph;
SELECT create_graph('network');
NOTICE:  graph "network" has been created
 create_graph 
--------------
 
(1 row)

--
-- inet
--
SELECT * FROM cypher('network', $$ RETURN toinet('192.168.1.5') $$) as (i gtype);
      i      
-------------
 192.168.1.5
(1 row)

SELECT * FROM cypher('network', $$ RETURN toinet('192.168.1/24') $$) as (i gtype);
       i        
----------------
 192.168.1.0/24
(1 row)

SELECT * FROM cypher('network', $$ RETURN toinet('::ffff:fff0:1') $$) as (i gtype);
         i          
--------------------
 ::ffff:255.240.0.1
(1 row)

SELECT * FROM cypher('network', $$ RETURN '192.168.1.5'::inet $$) as (i gtype);
      i      
-------------
 192.168.1.5
(1 row)

SELECT * FROM cypher('network', $$ RETURN '192.168.1/24'::inet $$) as (i gtype);
       i        
----------------
 192.168.1.0/24
(1 row)

SELECT * FROM cypher('network', $$ RETURN '::ffff:fff0:1'::inet $$) as (i gtype);
         i          
--------------------
 ::ffff:255.240.0.1
(1 row)

--
-- cidr
--
SELECT * FROM cypher('network', $$ RETURN tocidr('192.168.1.5') $$) as (i gtype);
       i        
----------------
 192.168.1.5/32
(1 row)

SELECT * FROM cypher('network', $$ RETURN tocidr('192.168.1/24') $$) as (i gtype);
       i        
----------------
 192.168.1.0/24
(1 row)

SELECT * FROM cypher('network', $$ RETURN tocidr('::ffff:fff0:1') $$) as (i gtype);
           i            
------------------------
 ::ffff:255.240.0.1/128
(1 row)

SELECT * FROM cypher('network', $$ RETURN '192.168.1.5'::cidr $$) as (i gtype);
       i        
----------------
 192.168.1.5/32
(1 row)

SELECT * FROM cypher('network', $$ RETURN '192.168.1/24'::cidr $$) as (i gtype);
       i        
----------------
 192.168.1.0/24
(1 row)

SELECT * FROM cypher('network', $$ RETURN '::ffff:fff0:1'::cidr $$) as (i gtype);
           i            
------------------------
 ::ffff:255.240.0.1/128
(1 row)

--
-- macaddr
--
SELECT * FROM cypher('network', $$ RETURN tomacaddr('12:34:56:78:90:ab') $$) as (i gtype);
         i         
-------------------
 12:34:56:78:90:ab
(1 row)

SELECT * FROM cypher('network', $$ RETURN '12:34:56:78:90:ab'::macaddr $$) as (i gtype);
         i         
-------------------
 12:34:56:78:90:ab
(1 row)

--
-- macaddr8
--
SELECT * FROM cypher('network', $$ RETURN tomacaddr8('12:34:56:78:90:ab:cd:ef') $$) as (i gtype);
            i            
-------------------------
 12:34:56:78:90:ab:cd:ef
(1 row)

SELECT * FROM cypher('network', $$ RETURN '12:34:56:78:90:ab:cd:ef'::macaddr8 $$) as (i gtype);
            i            
-------------------------
 12:34:56:78:90:ab:cd:ef
(1 row)

SELECT drop_graph('network', true);
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table network._ag_label_vertex
drop cascades to table network._ag_label_edge
NOTICE:  graph "network" has been dropped
 drop_graph 
------------
 
(1 row)

